### DISCUSSION TITLE
the ability to insert your own data into the simulation for Betfair

### DISCUSSION THREAD

**Discussion by AlesyaSeliazniova30032012** [nautechsystems/nautilus_trader#1478] [Q&A]
Hi. I wanted to test the action of your Nautilus trader simulator for Betfair. However, I do not have the opportunity to receive.bz2 file . Instead, I have a .csv file with the following data: ['Timestamp', 'RunnerId', 'runner_status', 'mkt_status', 'inPlay',
'adjustmentFactor', 'betDelay', 'bid', 'ask', 'bid_sz', 'ask_sz', 'ltp',
'tv', 'runner_name', 'timezone', 'TimestampLocal', 'MarketId', 'name',
'marketTime', 'marketTimeLocal', 'countryCode', 'eventId', 'eventName','openDate', 'openDateLocal', 'venue']. Is it possible to insert this historical data that I get using my own Betfair parsing into the code? And is there any documentation about this. Thanks

**Comment by AlesyaSeliazniova30032012** [comment]
no answer

**Comment by joeHollander** [comment]
Typically, it would involve creating a data class that inherits from Data. Then you could parse your csv into nautilus objects and then add it as data or publish to the message bus from an actor. Some related links below:

https://github.com/nautechsystems/nautilus_trader/discussions/425
https://github.com/nautechsystems/nautilus_trader/discussions/576

https://nautilustrader.io/docs/latest/concepts/advanced/custom_data/
https://nautilustrader.io/docs/latest/tutorials/loading_external_data/
